prac_06/languages.py

```diff
@@ -0,0 +1,22 @@
+"""
+Languages
+Estimated: 30 minutes
+Actual: 24 minutes
+"""
+
+from prac_06.programming_language import ProgrammingLanguage
+
+python = ProgrammingLanguage("Python", "Dynamic", True, 1991)
+ruby = ProgrammingLanguage("Ruby", "Dynamic", True, 1995)
+visual_basic = ProgrammingLanguage("Visual Basic", "Static", False, 1991)
+print(python)
+
+programming_languages = [ProgrammingLanguage("Python", "Dynamic", True, 1991),
+                         ProgrammingLanguage("Ruby", "Dynamic", True, 1995),
+                         ProgrammingLanguage("Visual Basic", "Static", False, 1991)]
+
+print("The dynamically typed languages are:")
+dynamic_languages = [programming_language.name for programming_language in programming_languages
+                     if programming_language.is_dynamic()]
+for i in range(len(dynamic_languages)):
+    print(dynamic_languages[i])
```

Comment:
This works and gives the desired output but I feel defeats the purpose of storing the languages in a class. 

The list comprehension could instead return the object and you could iterate over the list of languages and access their names e.g.

```Python
new_things = [thing for thing in old_things if thing.something()]  # return a list of objects
for thing in new_things:
    print(thing.name)  # grab the name from the object
```

Reply:
Accessing the object's name rather than using the list index also improves the readability of the code. Thanks! :)